tools/exe/palletjack2ipahost

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env ruby
+
+# Create host objects in a FreeIPA or Red Hat IdM instance.
+#
+# This script will loop over all hosts in the PalletJack database and check
+# if they are registered on the IPA server. The missing hosts will be added.
+# A random password will be generated by IPA and stored in the pallet.
+#
+# If the IPA server is configured as authorative DNS server, records will
+# be created for the new host.
+#
+# Data model assumptions:
+# - Each domain corresponds uniquely to one IPv4 network
+
+require 'palletjack/tool'
+require 'palletjack/ipa'
+require 'yaml'
+
+class PalletJack2Ipahost < PalletJack::Tool
+
+  def process
+    @ipa_pallets = {}
+    # Grab a list of all hosts in IPA
+    host_find = PalletJack::Ipa::Command.new("host_find")
+
+    if host_find.error? then
+      puts "ERROR: " + host_find.error_name + " " + host_find.error_message
+      exit 1
+    end
+
+    # Then convert the result to an array of FQDN strings
+    ipa_hosts = []
+    host_find.results.each do |host|
+      ipa_hosts << host['fqdn'][0]
+    end
+
+    jack.each(kind: 'system') do |system|
+      next if ipa_hosts.include? system['net.dns.fqdn'] # Skip existing hosts
+      next unless system['system.role'].include? 'ipa-client' # Only ipa clients
+
+      # Generate a random password
+      add_host_params = { 'random' => true } 
+
+      ip_address = nil
+      system.children(kind:'ipv4_interface') do |interface|
+        # Find a interface in the same domain as the host.
+        if interface['net.dns.domain'] == system['net.dns.domain'] then
+          ip_address = interface['net.ipv4.address']
+        end
```

Comment:
At the moment only the first IP address of the systems domain will get used. This may be a problem, but on the other hand if we are going to use IPA as authoritative DNS we probably want to create a separate tool to do a more complete sync of DNS records.